src/webots/gui/WbStreamingServer.cpp

```diff
@@ -78,10 +79,16 @@ void WbStreamingServer::setMainWindow(WbMainWindow *mainWindow) {
 }
 
 void WbStreamingServer::start(int port) {
+  mPort = port;
   try {
     create(port);
   } catch (const QString &e) {
     WbLog::error(tr("Error when creating the TCP streaming server on port %1: %2").arg(port).arg(e));
+    if ((WbPreferences::instance()->value("Streaming/port", 1234).toInt() + 10) > port) {
```

Comment:
What is the use of this line?

Reply:
The +10 is to prevent the recursive function to try to many times (if the 10th next ports cannot be opened we should not continue to try).